dspace-uclouvain/src/main/java/org/dspace/uclouvain/administer/OrgUnitImporter.java

```diff
@@ -0,0 +1,246 @@
+            }
+        }
+    }
+
+    /**
+     * Adds entity metadata on the newly created DSpace item.
+     * @param item: The Dspace item on which add metadata
+     * @param entity: The entity (from config file) to analyze
+     * @param parent: (optional) the parent Dspace item (to create parent-child relation)
+     * @throws SQLException if any database exception occurred during the process.
+     */
+    private void addEntityMetadata(Item item, Entity entity, Item parent) throws SQLException {
+        MetadataField mdField = null;
+        // Adds `entity.name` into `dc.title` metadata field.
+        mdField = metadataFieldService.findByString(this.context, "dc.title", '.');
+        itemService.addMetadata(this.context, item, mdField, null, entity.name);
+        // Adds `entity.acronym` into `oairecerif.acronym` metadata field.
+        if (entity.acronym != null) {
+            mdField = metadataFieldService.findByString(this.context, "oairecerif.acronym", '.');
+            itemService.addMetadata(this.context, item, mdField, null, entity.acronym);
+        }
+        // Adds `entity.type` into `dc.type` metadata field.
+        mdField = metadataFieldService.findByString(this.context, "dc.type", '.');
+        String mdValue = (entity.type != null) ? entity.type : this.defaultEntityType;
+        itemService.addMetadata(this.context, item, mdField, null, mdValue);
+        // Adds `entity.selectable` into `organization.isSelectable` metadata field.
+        mdField = metadataFieldService.findByString(this.context, "organization.isSelectable", '.');
+        itemService.addMetadata(this.context, item, mdField, null, String.valueOf(entity.selectable));
+        // Adds `entity.identifiers` into `organization.identifier.xxx`
+        if (entity.identifiers != null && !entity.identifiers.isEmpty()) {
+            for (Identifier identifier : entity.identifiers) {
+                String mdString = String.format("organization.identifier.%s", identifier.type);
+                mdField = metadataFieldService.findByString(this.context, mdString, '.');
+                itemService.addMetadata(this.context, item, mdField, null, identifier.value);
+            }
+        }
+        // Creates relation to parent if exists
+        if (parent != null) {
+            mdField = metadataFieldService.findByString(this.context, "organization.parentOrganization", '.');
+            itemService.addMetadata(this.context, item, mdField, null, parent.getName(), parent.getID().toString(), 600);
+        }
+    }
+
+    /**
+     * Search the 'OrgUnit' collection that will hold the entities
+     * @return the 'OrgUnit' parent collection
+     * @throws Exception if any error occurred during the search.
+     */
+    private Collection getOrgUnitCollection() throws Exception {
+        DiscoverQuery dq = new DiscoverQuery();
+        dq.setMaxResults(1);
+        dq.setQuery("search.resourcetype:Collection AND dc.title_sort:\"OrgUnit\"");
```

Comment:
NOTE: You could also search on 'dspace.entity.type' which is set to "orgunit" for a collection that handles orgunit entit type.
By doing so, you could give your collection any name you want and it will still work.
It really depends on if we will have one day multiple collection that are handling the type 'Orgunit'.